consensus/src/experimental/buffer_item.rs

```diff
@@ -31,39 +32,79 @@ fn generate_commit_proof(
     )
 }
 
-fn aggregate_ledger_info(
+fn get_verified_signatures(
+    unverified_signatures: &PartialSignatures,
+    validator: &ValidatorVerifier,
+    commit_ledger_info: &LedgerInfo,
+) -> PartialSignatures {
+    // Returns a valid partial signature from a set of unverified signatures.
+    // TODO: Validating individual signatures in expensive. Replace this with optimistic signature
+    // verification for BLS.
+    PartialSignatures::new(
+        unverified_signatures
+            .signatures()
+            .iter()
+            .filter(|(author, sig)| validator.verify(**author, commit_ledger_info, sig).is_ok())
+            .map(|(author, sig)| (*author, sig.clone()))
+            .collect(),
+    )
+}
+
+fn generate_executed_item_from_ordered(
+    commit_info: BlockInfo,
+    executed_blocks: Vec<ExecutedBlock>,
+    unverified_signatures: PartialSignatures,
```

Comment:
we should be able to verify all signatures here, I'd just filter invalid signatures out from any stages other than ordered